app/models/parameter.rb

```diff
@@ -10,6 +10,29 @@ class Parameter < ActiveRecord::Base
   validates :name, :presence => true, :no_whitespace => true
 
   scoped_search :on => :name, :complete_value => true
+  scoped_search :on => :type, :complete_value => true
+  scoped_search :on => :value, :complete_value => true
+
+  # children associations must be defined here, otherwise scoped search definitions won't find them
+  belongs_to :domain, :foreign_key => :reference_id, :inverse_of => :domain_parameters
```

Comment:
I agree it would make more sense. Unfortunately it wouldn't work as mentioned in comment above (should I rephrase it?). Scoped search [does not work](https://github.com/wvanbergen/scoped_search/issues/112) well with STI.

Getting this when I move both definitions to domain parameter and then trying to do `Domain.search_for('domain_name ~ example')`.
```
NoMethodError: undefined method `klass' for nil:NilClass
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/definition.rb:62:in `klass'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/query_builder.rb:302:in `to_sql'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/query_builder.rb:219:in `sql_test'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/query_builder.rb:514:in `sql_test'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/query_builder.rb:471:in `to_single_field_sql'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/query_builder.rb:483:in `to_sql'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/query_builder.rb:54:in `build_find_params'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/query_builder.rb:22:in `build_query'
        from /home/ares/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/scoped_search-3.3.0/lib/scoped_search/definition.rb:267:in `block in register_named_scope!'
```

Or did I misinterpret your suggestion?